app/controllers/api/labwares_controller.rb

```diff
@@ -30,20 +40,18 @@ def labwares_by_location
     location_barcodes = params[:location_barcodes]
     return unless location_barcodes
 
-    Labware.joins(:location).where(locations: { barcode: location_barcodes.split(',') })
+    Labware.by_location_barcode(location_barcodes.split(','))
   end
 
   def labwares_by_barcode
-    barcodes = params[:barcodes]
-    return [] unless barcodes
-
-    Labware.by_barcode(barcodes)
+    Labware.by_barcode(params[:barcodes])
   end
 
-  def labwares_by_barcode_known_locations
-    barcodes = params[:barcodes]
-    return [] unless barcodes
-
+  def labwares_by_barcode_known_locations(barcodes)
```

Comment:
With the refactoring you have done this method is now superfluous.

Reply:
Is it? The `by_barcode_known_locations` scope doesn't include any labwares with UnknownLocation, where as the `labwares_by_barcode` scope does. I'm also wondering whether the successful create response should actually return l`labwares_by_barcode` scope, which includes _all_ labwares, not just known ones?

Reply:
Maybe there is a misunderstanding. All the `labwares_by_barcode_known_locations` method does now is call `Labware.by_barcode_known_locations(barcodes)` I was just wondering whether we still need it?

Reply:
I assume so, the `by_barcode` method calls `labwares_by_barcode_known_locations` if `["known"] == "true"`, otherwise it calls `labwares_by_barcode`. 
What scope do you think the successful response to Labware creation should return? Will labwares ever be created via this new endpoint with a location of `UnknownLocation`?